source/jobs/JobsFeature.cpp

```diff
@@ -167,10 +173,19 @@ void JobsFeature::subscribeToUpdateJobExecutionStatusAccepted(string jobId)
         std::bind(
             &JobsFeature::updateJobExecutionStatusAcceptedHandler, this, std::placeholders::_1, std::placeholders::_2),
         std::bind(&JobsFeature::ackSubscribeToUpdateJobExecutionAccepted, this, std::placeholders::_1));
+
+    if (std::future_status::timeout == updateAcceptedPromise.get_future().wait_for(std::chrono::seconds(10)))
```

Comment:
Can't you use the promise returned by `SubscribeToUpdateJobExecutionAccepted` instead?

https://aws.github.io/aws-iot-device-sdk-js-v2/classes/aws_iot_device_sdk.iotjobsclient.html#subscribetoupdatejobexecutionaccepted